### PR TITLE
feat: Index vectors

### PR DESCRIPTION
**Pull request by attiks**
Refs: #RW-989

### CODE REVIEW

File: src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php [hunk 1]
```diff
@@ -80,34 +81,199 @@ public function getRelevantContents(string $index, array $ids, string $query_tex
...
+              ],
+            ],
+          ],
+          'embedding' => [
+            'type' => 'dense_vector',
+            'dims' => $dimensions,
+            'index' => TRUE,
+            'similarity' => 'dot_product',
+          ],
+          'contents' => [
```
Comment (orakili) [inline src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php:203]: Is that needed? There is only 1 embedding field for the whole document.

Comment (attiks) [reply src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php:203]: Can you clarify?

Comment (orakili) [reply src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php:203]: I meant that the `contents` property doesn't seem to be used anywhere. 

In the `Elasticsearch` plugin, this is used to store nested content (pages and for each page the chunk of texts with their embedding).

But for the `Elasticsearch Job` plugin, this doesn't seem to be used. There seems to be only embedding for the whole content (line 197).

So maybe the `contents` property can be remove from the mapping?

File: src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php [hunk 2]
```diff
@@ -45,69 +46,178 @@ public function getRelevantContents(string $index, array $ids, string $query_tex
...
+        // Retrieve the minimum similarity to be considered relevant.
+        $cutoff = $this->getSimilarityScoreCutOff($contents, .20);
+
+        // Exclude irrelevant contents.
+        return array_filter($contents, function ($score) use ($cutoff) {
+          return $score >= $cutoff;
+        });
+      }
+    }
+    catch (GuzzleException $exception) {
```
Comment (orakili) [inline src/Plugin/ocha_ai/VectorStore/ElasticsearchJob.php:90]: The catch added to the `::request()` method line 742 of `src/Plugin/ocha_ai/VectorStore/Elasticsearch.php` means that there won't be any GuzzleException thrown.

On the other hand, the `$response` form the `$this->request()` may be NULL in which PHP will complain line 64 when trying to get the body (`$response->getBody()`).

So I think this needs to be refactored a bit.

**Review by orakili — CHANGES_REQUESTED** [review]

**Review by orakili — APPROVED** [review]

**Pull request merged**